its/src/test/java/com/sonar/it/scanner/msbuild/utils/BuildCommand.java

```diff
@@ -40,21 +43,43 @@ public BuildCommand setTimeout(int timeout) {
   }
 
   public BuildCommand useDotNet() {
-    useDotNet = true;
+    return useDotNet("build");
+  }
+
+  public BuildCommand useDotNet(String dotnetCommand) {
+    this.dotnetCommand = dotnetCommand;
+    return this;
+  }
+
+  public BuildCommand addArgument(String... arguments) {
+    this.arguments.addAll(Arrays.asList(arguments));
     return this;
   }
 
   public BuildResult execute(Orchestrator orchestrator) {
     // ToDo: SCAN4NET-312: Extract createCommand, add support for parameters, add support for MsBuild vs dotnet. Should be similar to ScannerCommand.execute()
     // ToDo: SCAN4NET-312: Run restore & rebuild. Restore could be via NuGet (on request, or every time for simplicity?), MsBuild or dotnet
     var environmentVariables = environment.entrySet().stream().map(x -> new EnvironmentVariable(x.getKey(), x.getValue())).toList();
-    var result = useDotNet
-      ? TestUtils.runDotnetCommand(projectDir, environmentVariables, "build", "--no-incremental")
-      : TestUtils.runMSBuild(orchestrator, projectDir, environmentVariables, timeout, "/t:Restore,Rebuild");
+    var result = dotnetCommand == null
+      ? TestUtils.runMSBuild(orchestrator, projectDir, environmentVariables, timeout, prependArgument("/t:Restore,Rebuild"))
+      : TestUtils.runDotnetCommand(projectDir, environmentVariables, dotnetCommand, dotnetArguments());
```

Comment:
This whole thing, together with the new methods below, is ugly on purpose. The other sprint will work on BuildCommand